src/utils.js

```diff
@@ -365,11 +366,18 @@ export const guessPin = async (accessData, progressCb) => {
  *
  * @return {boolean} Wether the change password was successful
  */
-export const changePin = (oldPin, newPin) => (
-  hathorLib.wallet.changePinAndPassword({
+export const changePin = (oldPin, newPin) => {
+  const success = hathorLib.wallet.changePinAndPassword({
     oldPin,
     newPin,
     oldPassword: oldPin,
     newPassword: newPin,
-  })
-);
+  });
+  if (success) {
+    Keychain.setGenericPassword(KEYCHAIN_USER, newPin, {
```

Comment:
I feel we should create an util method in this same file `setKeychainPassword(pin)` that does this code, so we don't repeat this here and in the `startWallet` method. What do you think?